Gems/RecastNavigation/Code/Tests/MockInterfaces.h

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Component/Entity.h>
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/UnitTest/TestTypes.h>
+#include <AzFramework/Entity/EntityDebugDisplayBus.h>
+#include <AzFramework/Physics/PhysicsScene.h>
+#include <AzFramework/Physics/Shape.h>
+#include <AzTest/AzTest.h>
+#include <LmbrCentral/Shape/ShapeComponentBus.h>
+#include <RecastNavigation/RecastNavigationMeshBus.h>
+#include <AzCore/Time/ITime.h>
+#include <AzCore/Component/TransformBus.h>
+
+namespace RecastNavigationTests
+{
+    using namespace AZ;
+    using namespace AZStd;
+    using namespace testing;
```

Comment:
update: Please don't `using namespace` like this - it can cause some horrible problems, instead please use using declarations like `using AZ::<specificType>` where appropriate

Reply:
Done.